### PR TITLE
Back port helperWhiteList to 5.0.0

### PR DESCRIPTION
**Pull request by chadhietala**
/cc @stefanpenner See #95 for details

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -198,6 +204,32 @@ Babel.prototype.processString = function(string, relativePath) {
...
+
+          return item + '`, `';
+        }).join('');
+
+        message = relativePath + ' was transformed and relies on `' + helpers + '`, which were not included in the helper whitelist. Either add these helpers to the whitelist or refactor to not be dependent on these runtime helper.';
+      }
+
+      throw new Error(message);
+    }
+  }
```
Comment (Turbo87) [inline index.js]: would it make sense to extract that into a separate method/function to reduce the code complexity?

**Comment by Turbo87** [comment]
looks alright to me. what are you going to use this for?

**Comment by stefanpenner** [comment]
> looks alright to me. what are you going to use this for?

We want teams to be able to decide which helpers are allowed in their apps. Its often surprising to see some pretty bonkers code (although it is bonkers for good reason) in the output. So we want teams to be able to provide a whitelist, preventing unexpected output.

**Comment by stefanpenner** [comment]
@Turbo87 We also want the ability tor teams (like ember or w/e) to provide there own helpers, once that are more tailored for the use-case or the subset of functionality we need. Often the trade-off (perf/bytes etc) requires this.

**Comment by stefanpenner** [comment]
@chadhietala this LGTM

**Comment by rwjblue** [comment]
I discussed with @chadhietala yesterday, but I'd like to see this code take advantage of the dep graph stuff that is already here.

IMHO, a better user experience for this would be to perform all of the transpilations, and emit a single error that listed all of the helpers and files that include non-whitelisted helpers.  As things stand right now, it would be a giant game of whack-a-mole 😝 .

**Comment by stefanpenner** [comment]
> global

per instance of babel right?

**Comment by rwjblue** [comment]
@stefanpenner - Ya, I think it would be per input tree being compiled or something.  Haven't fully specc'ed it out or anything, just saying that tracking the helpers used as part of the larger dep graph stuff would unblock some things that are pretty difficult to do outside of broccoli-babel-transpiler.

**Comment by chadhietala** [comment]
How shall we proceed?

**Pull request merged**